### ISSUE TITLE
TapToPlace Overwrites Surface Normal Offset

### ISSUE THREAD

**Issue by jbienzms** [microsoft/MixedRealityToolkit-Unity#8053]
## Describe the bug

No matter what value is specified in the inspector, when **Start** is called in **TapToPlace.cs**, this value is overwritten with the bounds size. 

## To reproduce

1. Add a **BoxCollider** to an object. Set the **Size** to 0.3, 0.2, 0.3.
2. Add **SolverHandler** and **TapToPlace** to the same object.
3. In the inspector for **TapToPlace**, set **Surface Normal Offset** to **0.001**.
4. Run the app and notice that the **Surface Normal Offset** has been replaced with **0.1500001**

## Expected behavior

I could understand possibly overwriting this value if none had been specified. But if if the user specifies a value, I don't think it should be overwritten.

## Your setup (please complete the following information)

- Unity Version 2019.2.21f1
- MRTK Version v2.4

## Target platform (please complete the following information)

- HoloLens 2

**Comment by CDiaz-MS** [comment]
The SurfaceNormalOffset is set to the z extents of the bounds on the collider by default (this is how the 0.1500001 is calculated).  This offset is the default to prevent half of the object from being placed outside of a wall because the transform point is the center of a game object.  But if a user sets a different value, it should not be overwritten.

**Comment by jbienzms** [comment]
> But if a user sets a different value, it should not be overwritten.

Agreed. But if you look at TapToPlace [Line 200](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/375f306d0af44e26335297b2fc988f4d612bf787/Assets/MRTK/SDK/Features/Utilities/Solvers/TapToPlace.cs#L200) you'll notice that it executes unconditionally. This is what causes the bug.

Since this is a floating point value there's not really any way to determine if the current setting is "unspecified". The user may actually _want_ to specify 0.0f. Probably the only thing to do without major refactoring would be to consider 0.0F as unspecified and then use the bounds. If the user then wants to specify no offset, they could specify something like 0.0001F. This should be documented, however, to make it clear.

For additional background: In my case I do want the offset to be 0.0F. I'm using an arrow to point at locations on the floor and I want the arrow to touch the floor instead of hovering over it. I've locked the arrows orientation vertically and it's OK if the arrow clips into the wall.

**Comment by CDiaz-MS** [comment]
I think adding a new `UseDefaultSurfaceNormalOffset` bool to the component could make things clearer.  If `UseDefaultSurfaceNormalOffset` is true, then the z extents of the bounds will be used.  If it's false, then the value for `SurfaceNormalOffset` will be used.  This way people can use 0 as a `SurfaceNormalOffset` if needed.

I can also add a visibility condition in TapToPlace's inspector, if `UseDefalutSurfaceNormalOffset` is true, then do not display a field for `SurfaceNormalOffset`.  If false, display it.  This will all of course be code configurable.

**Comment by jbienzms** [comment]
That would be clearer, yes. I sort of hate to add a whole property for that, but it would be clear without needing to call out the otherwise somewhat "hidden" behavior in documentation.
